Modules/Core/Common/include/itkMacro.h

```diff
@@ -1386,6 +1386,16 @@ itkDynamicCastInDebugMode(TSource x)
 #endif
 }
 
+#ifdef ITK_LEGACY_REMOVE
+#  if __cplusplus >= 202002L
+#    define ITK_NO_DISCARD(message) [[nodiscard(#    message)]]
+#  else
+#    define ITK_NO_DISCARD(message) [[nodiscard]]
+#  endif
+#else
+#  define ITK_NO_DISCARD(message)
+#endif
```

Comment:
Thanks @dzenanz Some suggestions:
- For consistency with C++20 `[[nodiscard("Some message")]]` uses, I would suggest to remove the `#` and simply always add double-quotes to the actual message when _calling_ the macro.
- For consistency with the C++ attribute name, maybe just do `NODISCARD` without underscore.

Reply:
Applied the first two suggestions. I leave the third to you 😄

Reply:
> I leave the third to you

Maybe just wait for ITK to upgrade to C++20 😸